File: 02/part-1.py
with open("./input.txt", "r") as f:
	rounds = [ln.split(" ") for ln in f.read().split("\n") if ln]


def win(round: [int, int]):
	"""
	-1 : loose
	 0 : draw
	 1 : win
	"""
	if (round[0] == "A"):
		if round[1] == "Z": return -1
		elif round[1] == "Y": return 1
	elif (round[0] == "B"):
		if round[1] == "X": return -1
		elif round[1] == "Z": return 1
	elif (round[0] == "C"):
		if round[1] == "Y": return -1
		elif round[1] == "X": return 1

	return 0


def getScore(win: bool, played: str):
	score = 0
	if (win == 0): score += 3
	elif (win == 1): score += 6

	score += {"X": 1, "Y": 2, "Z": 3}[played]

	return score


sum = 0
for round in rounds:
	result = win(round)
	score = getScore(result, round[1])

	print(round, result, score)

	sum += score

print("=>", sum)

File: 03/part-1.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

int getScore(char c)
{
    if (c > 64 && c < 91)
        return c - 38;
    else
        return c - 96;
}

int main()
{
    FILE *input = fopen("./03/input.txt", "r");
    if (input == NULL)
    {
        printf("Could not open input");
        exit(1);
    }

    unsigned char line[200];
    int sum = 0;
    while (!feof(input))
    {
        fgets(line, 200, input);
        int len = strlen(line) - 1;

        unsigned char count[128] = {0};
        unsigned char issue = 0;

        for (int i = 0; i < len; i++)
        {
            if (i >= len / 2 + 1)
            {
                if (count[line[i]] > 0)
                {
                    issue = line[i];
                    break;
                }
            }
            else
            {
                count[line[i]]++;
            }
        }
        int score = getScore(issue);
        sum += score;
        printf("%c : %d\n", issue, score);
    }

    fclose(input);
    printf("=> %d\n", sum);
    return 0;
}
